Add Conditional PostgreSQL Support to Moodle Helm Chart

Awungia112 opened the pull request:
This PR adds optional PostgreSQL support to the Moodle Helm chart by including Bitnami's PostgreSQL chart as a conditional dependency.

### Key Changes

- Added `bitnami/postgresql` as a chart dependency in `Chart.yaml`
- Made the PostgreSQL deployment conditional via `values.yaml`
- Updated `values.yaml` with default values 
- Created a comprehensive `README.md` with configuration instructions and local testing steps
- Verified the chart using:
  - `helm template` dry run
  - Local K3s deployment

Awungia112 committed:
Update Chart.lock for Moodle Helm chart dependencies

- Replace app-template dependency with moodle and postgresql
- Update versions for moodle (27.0.3) and postgresql (12.5.7)
- Regenerate lock file with new digest and timestamp

Awungia112 committed:
Update Moodle Helm chart version and add PostgreSQL dependency

- Bump appVersion to 5.0 for Moodle
- Add PostgreSQL dependency with version 12.5.7 and repository URL

Awungia112 committed:
Add README for Moodle Helm Chart

- Introduced a comprehensive README file detailing the Moodle Helm chart
- Included features, usage instructions, testing guidelines, and dependencies
- Provided installation and uninstallation commands for ease of use

Awungia112 committed:
Add PostgreSQL configuration and Moodle database settings to values.yaml

- Enabled PostgreSQL with authentication details for Moodle
- Configured database connection settings including host, name, user, and password
- Added default Moodle user credentials for initial setup

stephane-segning commented on charts/moodle/values.yaml:
Postgresql should disabled per default

stephane-segning replied:
And add instead a `values-postgres.yaml` that will enable postgres

stephane-segning replied:
https://helm.sh/docs/chart_template_guide/values_files/

Awungia112 committed:
Update README for Moodle Helm Chart with enhanced installation options and database integration details

- Added sections for external and internal PostgreSQL installation options
- Clarified usage instructions and provided additional context for database configurations
- Improved formatting and organization for better readability

Awungia112 committed:
Add PostgreSQL values configuration for Moodle Helm Chart

- Introduced a new values-postgres.yaml file for PostgreSQL settings
- Configured PostgreSQL authentication and database details for Moodle
- Set default Moodle user credentials for initial setup

Awungia112 committed:
Disable PostgreSQL in Moodle Helm Chart configuration

Awungia112 committed:
Merge branch 'main' into feat/add-postgresql-dependency

stephane-segning commented on charts/moodle/values.yaml:
```diff
@@ -12,4 +12,19 @@ moodle:
+moodleDatabaseHost: postgresql
+moodleDatabaseName: bitnamimoodle
+moodleDatabaseUser: moodle
+moodleDatabasePassword: moodlepassword
```
In the default `values.yaml`, this is useless, please remove it

stephane-segning reviewed (changes requested):
Awesome work. one last comment:

Awungia112 committed:
Remove PostgreSQL database configuration from Moodle Helm Chart values.yaml
